yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnableUtil.scala

```diff
@@ -75,8 +75,9 @@ trait ExecutorRunnableUtil extends Logging {
     // registers with the Scheduler and transfers the spark configs. Since the Executor backend
     // uses Akka to connect to the scheduler, the akka settings are needed as well as the
     // authentication settings.
-    sparkConf.getAll.
-      filter { case (k, v) => k.startsWith("spark.auth") || k.startsWith("spark.akka") }.
+    sparkConf.getAll.filter { case (k, v) =>
+      k.startsWith("spark.auth") || k.startsWith("spark.akka") || k.equals("spark.port.maxRetries")
+    }.
```

Comment:
The crux of the problem here seems to be that YARN executors receive some config settings via System properties as JVM flags, and other config settings via the SparkEnv object they receive after getting connected up to the rest of the cluster.

This whitelist is for passing config options that must be received by the executor before connection, and `spark.port.maxRetries` seems to be one of those indeed.